packages/client/src/types.ts

```diff
@@ -0,0 +1,373 @@
+import {
+  BeforeHookContext,
+  ClientMetadata,
+  CommonProvider,
+  EvaluationContext,
+  EvaluationDetails,
+  Eventing,
+  FlagValue,
+  HookContext,
+  HookHints,
+  JsonValue,
+  Logger,
+  ManageContext,
+  ManageLogger,
+  ManageTransactionContextPropagator,
+  ProviderMetadata,
+  ResolutionDetails,
+} from '@openfeature/shared';
+
+/**
+ * Interface that providers must implement to resolve flag values for their particular
+ * backend or vendor.
+ *
+ * Implementation for resolving all the required flag types must be defined.
+ */
+export interface Provider extends CommonProvider {
+
+  /**
+   * A provider hook exposes a mechanism for provider authors to register hooks
+   * to tap into various stages of the flag evaluation lifecycle. These hooks can
+   * be used to perform side effects and mutate the context for purposes of the
+   * provider. Provider hooks are not configured or controlled by the application author.
+   */
+  readonly hooks?: Hook[];
+
+  // client vs global context?
+  onContextChange?(oldContext: EvaluationContext, newContext: EvaluationContext): Promise<void>
+
+  // TODO: move to common Provider type when we want close in server
+  onClose?(): Promise<void>;
+
+  /**
+   * Resolve a boolean flag and its evaluation details.
+   */
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context: EvaluationContext,
```

Comment:
I'm suprised that we're passing context here. I thought that the static context paradigm would mean you can't pass context at the point of evaluation, in exchange for it being a synchronous call.